src/v2/System/Router/RenderStatus.tsx

```diff
@@ -77,14 +77,15 @@ export const RenderError: React.FC<{
   }
 
   const message =
-    (process.env.NODE_ENV || sd.NODE_ENV) === "development"
+    getENV("NODE_ENV") === "development"
       ? String(props.error.data)
-      : "Internal error"
+      : "Internal Error"
```

Comment:
Adding right after this line 
```tsx
if (typeof window === "undefined") {
    throw new HttpError(props.error.status, message)
  }
```
will yield proper status and pass things along to errorMiddleware

Reply:
(In any case, this `RenderStatus` is really more of a client-side thing, so i don't find it to be too weird, as long as the gotcha here has a comment)